request.js

```diff
@@ -1413,6 +1413,28 @@ Request.prototype.jar = function (jar) {
   return self
 }
 
+Request.prototype.cloneJar = function (cookieJar) {
+  var self = this
+  var newJar, cookieString, cookieStore, domains, domainPaths, cookieUrl
+
+  newJar = self.jar()
+  cookieStore = cookieJar._jar.store.idx
```

Comment:
Let me try some stuff, I'll let you know, I didn't want to iterate over the idx keys either but so far it has proven the only useful way. Passing a store to the request.jar() method gives a tough-cookie error at some point where the store is asyncronous.